CarePartner/Views/AccountSettingsView.swift

```diff
@@ -6,66 +6,196 @@
 //
 
 import SwiftUI
+import TidepoolKit
 
-struct AccountSettingsView: View {
-    @Environment(\.dismiss) var dismiss
+public struct AccountSettingsView: View {
+    @Environment(\.dismiss) private var dismiss
+    @EnvironmentObject var sceneDelegate: SceneDelegate
 
-    var accountLogin: String
-    var didRequestDelete: () -> Void
+    @State private var isEnvironmentActionSheetPresented = false
+    @State private var showingDeletionConfirmation = false
 
-    @State private var showingAlert = false
+    @State private var error: Error?
+    @State private var isLoggingIn = false
+    @State private var selectedEnvironment: TEnvironment
+    @State private var environments: [TEnvironment] = [TEnvironment.productionEnvironment]
+    @State private var environmentFetchError: Error?
 
-    var body: some View {
-        NavigationView {
-            VStack {
-                Text("Tidepool ")
-                    .font(.largeTitle)
-                    .fontWeight(.semibold)
-                Image(decorative: "Tidepool Logo")
-                    .resizable()
-                    .aspectRatio(contentMode: .fill)
-                    .frame(width: 150, height: 150)
+    @ObservedObject private var client: TidepoolClient
 
+    init(client: TidepoolClient)
+    {
+        self.client = client
+        let defaultEnvironment = client.api.defaultEnvironment
+        self._selectedEnvironment = State(initialValue: client.session?.environment ?? defaultEnvironment ?? TEnvironment.productionEnvironment)
```

Comment:
Can you not assign directly here?

Reply:
Because it's a property wrapper wrapped attribute, assignment triggers usage of the attribute before initialization, and we get "Variable 'self.selectedEnvironment' used before being initialized"